Sources/Core/Public/GTMSessionFetcher/GTMSessionFetcher.h

```diff
@@ -511,6 +511,41 @@ typedef void (^GTMSessionFetcherTestResponse)(NSHTTPURLResponse *_Nullable respo
 typedef void (^GTMSessionFetcherTestBlock)(GTMSessionFetcher *fetcherToTest,
                                            GTMSessionFetcherTestResponse testResponse);
 
+// Provides access to a user-agent string calculated on demand.
+@protocol GTMUserAgentProvider <NSObject>
```

Comment:
Shouldn't this document that implementor need to make any class conforming to this thread safe?  i.e. - could multiple fetchers ended up trying to call atleast the userAgentCached property while another thread might have already called userAgent?

Reply:
Added docs.